### PR TITLE
Rabby dev

### PR DESCRIPTION
**Pull request by R313mon**
1. Thumbnails now display rather than videos
2. Creators page can be accessed through video page
3. Users can upload banner and profile pictures to their accounts
4. User profile and creators page videos are routed to video page

### CODE REVIEW

File: emuu-client/src/components/CreatorsPage/CreatorsPage.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  useEffect(async () => {
+    if (video) {
+      localStorage.setItem("video", JSON.stringify(video));
+    }
+    if (localStorage.getItem("video")) {
+      let video = JSON.parse(localStorage.getItem("video"));
+      setCreatorName(video.Username);
+    }
+    if (!video && !localStorage.getItem("video")) {
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/CreatorsPage/CreatorsPage.js:39]: Don't think this is necessary for the creator's page, why would we redirect to home page if there's no video?

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+import "./Feeds.scss";
+import { Avatar } from "@mui/material";
+import YouTubeJSON from "../data/youtube-videos.json";
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/CreatorsPage/Feeds.js:4]: why is this here? We don't need placeholder YouTube videos anymore after the first prototype.

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    const recentVideosArr = [];
+    querySnapshotRecent.forEach((doc) => {
+      recentVideosArr.push(doc.data());
+    });
+
+    setRecentVideos(recentVideosArr);
+  }
+
+  //Sort function for date uploaded
+  function sortVideosByTime(videos) {
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/CreatorsPage/Feeds.js:41]: Good use of using the sort function for the profile page. Shows most recently uploaded on top.

File: emuu-client/src/components/UserProfile/Feeds.js [hunk 4]
```diff
@@ -14,16 +14,16 @@ import {
...
+function Feeds({ setVideo }, { setUserProfile }) {
   const [recentVideos, setRecentVideos] = useState([]);
   const displayName = localStorage.getItem("displayName");
 
   async function getVideos() {
     //Get all video data
-    const docRef = collection(db, "Users", displayName, "Videos");
-    const querySnapshot = await getDocs(docRef);
-
+    const docRef = collection(db, "Videos");
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Feeds.js:24]: Good job of getting rid of the sub collection for users. It was unnecessary to store the same data twice and update them twice.

File: emuu-client/src/components/upload/UploadButton.js [hunk 5]
```diff
@@ -27,24 +29,25 @@ const theme = createTheme({
     },
   },
 });
-function LinearProgressWithLabel(props: LinearProgressProps & { value: number }) {
+function LinearProgressWithLabel(
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/upload/UploadButton.js:32]: Good job of adding the linear progress bar instead of circular.

File: emuu-server/users/upload.go [hunk 6]
```diff
@@ -82,27 +82,8 @@ func UploadVideo(c *gin.Context) {
...
-                "usersThatLiked": usersThatLikedArr,
-                   "thumbnailUrl": input.Thumbnail_url,
-
-                })
-
-                if err != nil {
-                        log.Fatalf("firestore doc creation error:%s\n", err)
-                }
-        fmt.Println(wr.UpdateTime, uc.UpdateTime)
+        fmt.Println(wr.UpdateTime, wr.UpdateTime)
```
Comment (Tanvir-Rahman123) [inline emuu-server/users/upload.go:86]: Get rid of fmt.Println statements before merge to main.

**Review by Tanvir-Rahman123 — COMMENTED** [review]
Good work for prototype 2. Update some CSS if possible.

File: emuu-client/src/components/CreatorsPage/Feeds.js (hunk 2 above)
Comment (Moamin1) [inline emuu-client/src/components/CreatorsPage/Feeds.js:4]: We are no longer using the youtube-json file

**Review by Moamin1 — APPROVED** [review]
Good work overall, I don't see many issues.
